src/graph.rs

```diff
@@ -883,6 +883,12 @@ impl<N, E, Ty, Ix> Graph<N, E, Ty, Ix>
         &self.edges
     }
 
+    /// Return the internal arrays, consuming the graph.
+    pub fn destructure(self) -> (Vec<Node<N, Ix>>, Vec<Edge<E, Ix>>)
```

Comment:
This is my main hold up, since it seems to make the data structure more naked (less encapsulation).

Yet that's not really the case, we could make Node and Edge vectors from any graph representation. The existing raw_nodes etc accessor is a real encapsulation blocker in comparison. So this is good.